frontend/webservice/admin/admin.cgi

```diff
@@ -2559,16 +2567,52 @@ sub decom_workgroup{
     my ($method, $args) = @_;
 
     # my ($user, $err) = authorization(admin => 1, read_only => 0);
+    #Intial check to see if user is a normal system admin
     my ($result, $err) = OESS::DB::User::has_system_access(db => $db2, username => $ENV{'REMOTE_USER'}, role => 'normal'); 
     if (defined $err) {
         $method->set_error($err);
         return;
     }
-
+    #Find the workgroup to be decommissioned
     my $workgroup_id = $args->{'workgroup_id'}{'value'};
-    my $results;
+    my $workgroup = new OESS::Workgroup(db => $db2, workgroup_id => $workgroup_id);
+
+    if (!defined $workgroup) {
+        $method->set_error("No workgroup with that ID found");
+        return;
+    }
+    #If the workgroup to be decommissioned is an admin type workgroup escalate the permissions needed to high admin
+    if ($workgroup->{type} eq 'admin'){
+        ($result, $err) = OESS::DB::User::has_system_access(db => $db2, username => $ENV{'REMOTE_USER'}, role => 'admin');
+        if (defined $err) {
+            $method->set_error($err);
+            return;
+        }
+    }
+    # Gather interfaces to remove the acls and start the transaction
+    my $interfaces = OESS::DB::Interface::get_interfaces(db => $db2, workgroup_id => $workgroup_id);
+
+    $db2->start_transaction();
+
+    foreach my $interface (@$interfaces) {
+        my ($count, $acl_error) = OESS::DB::ACL::remove_all(db => $db2, interface_id => $interface);
+        if (defined $acl_error) {
+            $method->set_error($acl_error);
+            $db2->rollback();
+            return;
+        }
+    }
+    $db2->commit();
```

Comment:
If you call `commit` after `$workgroup->update` the entire change can be within a single transaction resulting in an all-or-nothing db change, which i think is preferable.